src/NServiceBus.Core/Serializers/XML/Deserializer.cs

```diff
@@ -274,7 +274,14 @@ object GetObjectOfTypeFromNode(Type t, XmlNode node)
                     type = Type.GetType("System." + n.Name.Substring(0, n.Name.IndexOf(":")), false, true);
                 }
 
-                var prop = GetProperty(t, n.Name);
+                Tuple<FieldInfo[], PropertyInfo[]> typeMembers;
+                if (!cache.typeMembers.TryGetValue(t, out typeMembers))
+                {
+                    cache.InitType(t);
```

Comment:
as it looks, the v5 code also didn't use runtime initialization when the type wasn't initialized already?